src/parser/smt2/smt2_lexer.cpp

```diff
@@ -342,6 +352,39 @@ Token Smt2Lexer::tokenizeCurrentSymbol() const
         return Token::INDEX_TOK;
       }
       break;
+    case '-':
+    {
+      // note that `-4`, `-4.0`, `-4/5` are SMT-LIB symbols, hence we only
+      // convert these to literals if we are not strict parsing.
```

Comment:
Why?

Reply:
Since according SMT-LIB, `-4` is a possible name for a user symbol.